ansible/library/conn_graph_facts.py

```diff
@@ -343,10 +343,6 @@ def main():
         results = {k: v for k, v in locals().items()
                    if (k.startswith("device_") and v)}
 
-        # flatten the lists for single host
-        if m_args['hosts'] is None:
-            results = {k: v[0] if isinstance(v, list) else v for k, v in results.items()}
-
```

Comment:
So for single-dut testbed, will the removal of flattening be a problem to `add-topo` or `remove-topo`?

Reply:
@lolyu I also had to change the ansible playbook yml files to reach one level down. There is also a failure at root fanout configure that doesn't seem related to this change. I have a fix for that but I have to delay putting it up until this PR is closed.